portal/src/main/scala/cromwell/pipeline/datastorage/dao/entry/UserEntry.scala

```diff
@@ -29,6 +29,12 @@ class UserEntry(val pipelineDatabaseEngine: PipelineDatabaseEngine) {
     users.filter(_.email === email).take(1)
   }
 
-  def addUserAction(user: User) = (users.returning(users.map(_.userId))) += user
+  def getUsersByEmailAction(emailPattern: String) = {
+    val query = for {//TODO - can we make it compiled?
+      user <- users if user.email like "%"+emailPattern+"%" //TODO - is it safe to give parameters to DB this way(to prevent SQL-injections)? what syntax should we use to concatenate strings? any preferred formaters?
```

Comment:
I'm not sure about Slick, but in general, this is not safe.
To format string you can use [string interpolation](https://docs.scala-lang.org/overviews/core/string-interpolation.html) "%$emailPattern%"

Reply:
users.filter(_.email like s"%$emailPattern%").result 
This works fine, thank you so much